src/ripple/app/tests/Env.h

```diff
@@ -0,0 +1,249 @@
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_TESTS_ENV_H_INCLUDED
+#define RIPPLE_APP_TESTS_ENV_H_INCLUDED
+
+#include <ripple/app/tests/Common.h>
+#include <ripple/app/tests/JTx.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/json/json_value.h>
+#include <ripple/json/to_string.h>
+#include <ripple/protocol/Issue.h>
+#include <ripple/protocol/RippleAddress.h>
+#include <ripple/protocol/STAmount.h>
+#include <ripple/protocol/STObject.h>
+#include <ripple/protocol/STTx.h>
+#include <beast/unit_test/suite.h>
+#include <boost/logic/tribool.hpp>
+#include <utility>
+#include <unordered_map>
+
+namespace ripple {
+namespace test {
+
+/** A view to an account's account root. */
+class AccountInfo
+{
+private:
+    Account account_;
+    std::shared_ptr<Ledger> ledger_;
+    std::shared_ptr<STLedgerEntry> root_;
+
+public:
+    AccountInfo(Account const& account,
+            std::shared_ptr<Ledger> const& ledger)
+        : account_(account)
+        , ledger_(ledger)
+        , root_(ledger->getAccountRoot(
+            account.id()))
+    {
+    }
+
+    STAmount
+    balance (Issue const& issue) const;
+
+    std::uint32_t
+    seq() const;
+
+    std::uint32_t
+    flags() const;
+};
+
+//------------------------------------------------------------------------------
+
+/** A transaction testing environment. */
+class Env
+{
+public:
+    beast::unit_test::suite& test;
+
+    /** The master account. */
+    Account const master;
+
+    /** The open ledger. */
+    std::shared_ptr<Ledger> ledger;
+
+public:
+    Env (beast::unit_test::suite& test_);
+
+    /** Associate AccountID with account. */
+    void
+    memoize (Account const& account);
+
+    /** Returns the Account given the AccountID. */
+    /** @{ */
+    Account const&
+    lookup (std::string const& base58ID) const;
+
+    Account const&
+    lookup (ripple::Account const& id) const;
+    /** @} */
+
+    /** Returns info on an Account. */
+    /** @{ */
+    AccountInfo
+    info (Account const& account) const
+    {
+        return AccountInfo(account, ledger);
+    }
+
+    AccountInfo
+    operator[](Account const& account) const
+    {
+        return info(account);
+    }
+    /** @} */
+
+    void auto_fee (bool value)
+    {
+        fill_fee_ = value;
+    }
+
+    void auto_seq (bool value)
+    {
+        fill_seq_ = value;
+    }
+
+    void auto_sig (bool value)
+    {
+        fill_sig_ = value;
+    }
+
+    /** Create a JTx from parameters. */
+    template <class... Args>
+    JTx
+    tx (Json::Value jv, Args const&... args)
```

Comment:
`args` are really functions that will be called, not arguments to functions. Maybe rename `args` -> `funs`.